Fix build for GCC15.

manfredsc opened the pull request:
Builds fine using this small patch on top of clang_wne. Some malloc/realloc warnings remain elsewhere, but I
had not yet time to look into it.

Fixes #309 in the configuration
  -DUSE_OPENMP=ON \
  -DUSE_REGEX=ON \
  -DUSE_IPOLATES=ON \
  -DUSE_AEC=ON \
  -DUSE_PNG=ON \
  -DUSE_G2CLIB=ON \
  -DUSE_OPENJPEG=ON \
  -DUSE_TIGGE=ON \
  -DMAKE_FTN_API=ON

manfredsc committed:
Fix build for GCC15.

manfredsc committed:
Merge branch 'NOAA-EMC:clang_wne' into clang_wne

AlysonStahl-NOAA commented:
Thank you for all of these contributions! 

Are all of these pull requests you made completed? They aren't marked as drafts, but I just wanted to double check before I review and merge the changes.

Also, commenting here since this merges into Wesley's branch. @webisu I'm not sure if you have permission to merge, but either way I want to make sure you're ready to merge this into your branch?

manfredsc commented:
- This PR: completed and tested.
- PR 338: completed and tested. WNE argued, that one could disable state plane projections. Can be done later.
- PR 339: completed and tested. Style of silencing compiler can be discussed, but works this way.
    I tracked the convoluted code paths and made sure that original code is correct for all treated -Wmaybe-uninitialized
    issues, so they are only cosmetic (Exception: Issue 340).
    Unfortunately I made the branch on top of the other branch, so the 2 patches from PR 338 are duplicated.
- PR 341:
   * wmo scripts and tables are completed and OK. There is one script I moved away from wmo_scripts, can be
      deleted later if desired.
  * gribtables/ncep: scripts and tables are completed and OK. Perhaps WNE could glance over the resulting table.
      Several new entries, one entry got renamed (CDNETS->CNETS).
  * gribtables/ecmwf: whole thing is at maximum of beta quality, but it is better than before. There are fundamental
     issues, I will try to document them a bit in the PR description. I suggest to merge it as-is.

manfredsc commented:
@AlysonStahl-NOAA : my fix did not make it into develop yet. Should I do a separate PR against the develop branch?

AlysonStahl-NOAA commented:
Yes, if I merge this it will just merge to Wesley's branch, not develop. Wesley's changes on his branch were already merged into develop.

manfredsc commented:
Fixed by #352 and now obsolete. Closing.
